### PR TITLE
Better documentation of queuing parameters

### PR DESCRIPTION
**Pull request by abidlabs**
Inspired by this [internal conversation](https://huggingface.slack.com/archives/C02QZLG8GMN/p1671090431731269), improved the documentation of certain parameters in queuing

### CODE REVIEW

File: gradio/blocks.py [hunk 1]
```diff
@@ -1260,7 +1261,7 @@ def launch(
             server_name: to make app accessible on local network, set this to "0.0.0.0". Can be set by environment variable GRADIO_SERVER_NAME. If None, will use "127.0.0.1".
             show_tips: if True, will occasionally show tips about new Gradio features
             enable_queue: DEPRECATED (use .queue() method instead.) if True, inference requests will be served through a queue instead of with parallel threads. Required for longer inference times (> 1min) to prevent timeout. The default option in HuggingFace Spaces is True. The default option elsewhere is False.
-            max_threads: allow up to `max_threads` to be processed in parallel. The default is inherited from the starlette library (currently 40).
+            max_threads: the maximum number of total threads that the Gradio app can generate in parallel. The default is inherited from the starlette library (currently 40). Applies whether the queue is enabled or not. But if queuing is enabled, this parameter is increaseed to be at least the concurrency_count of the queue.
```
Comment (freddyaboulton) [inline gradio/blocks.py:1264]: Wonder if this is useful to add to the "maximum performance" guide

**Review by freddyaboulton — APPROVED** [review]
Thanks for the clarification @abidlabs !

File: gradio/blocks.py (hunk 1 above)
Comment (abidlabs) [reply gradio/blocks.py:1264]: Great suggestion @freddyaboulton, added

File: gradio/blocks.py [hunk 2]
```diff
@@ -1184,32 +1185,32 @@ def clear(self):
...
-        max_size: Optional[int] = None,
+        max_size: int | None = None,
     ):
         """
         You can control the rate of processed requests by creating a queue. This will allow you to set the number of requests to be processed at one time, and will let users know their position in the queue.
         Parameters:
-            concurrency_count: Number of worker threads that will be processing requests concurrently.
+            concurrency_count: Number of worker threads that will be processing requests from the queue concurrently. Increasing this number will increase the rate at which requests are processed, but will also increase the memory usage of the queue.
             status_update_rate: If "auto", Queue will send status estimations to all clients whenever a job is finished. Otherwise Queue will send status at regular intervals set by this parameter as the number of seconds.
-            client_position_to_load_data: Once a client's position in Queue is less that this value, the Queue will collect the input data from the client. You may make this smaller if clients can send large volumes of data, such as video, since the queued data is stored in memory.
```
Comment (aliabd) [inline gradio/blocks.py:1198]: client_position_to_load_data is not documented in the docstring


Screen Shot 2022-12-15 at 7 13 56 PM

File: guides/06_other_tutorials/setting_up_a_demo_for_maximum_performance.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Setting Up a Demo for Maximum Performance
 
+Tags: QUEUE, PERFORMANCE
```
Comment (aliabd) [inline guides/06_other_tutorials/setting_up_a_demo_for_maximum_performance.md:3]: Just a note that this won't get the guide to show up under `queue` or `Interface`. The right way to get it to show up with these is to add it to the `Guides: ` section of the docstring for Interface.

Comment (abidlabs) [reply guides/06_other_tutorials/setting_up_a_demo_for_maximum_performance.md:3]: Yeah, that's fine, this was just for the tags

File: gradio/blocks.py (hunk 2 above)
Comment (abidlabs) [reply gradio/blocks.py:1198]: Fixed now

**Comment by abidlabs** [comment]
Thanks for the feedback @freddyaboulton and @aliabd!

**Pull request merged**